pkg/apis/postgres-operator.crunchydata.com/v1beta1/shared_types.go

```diff
@@ -0,0 +1,24 @@
+/*
+ Copyright 2021 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1beta1
+
+type ServiceSpec struct {
```

Comment:
:+1: Pulling shared types into its own file

Reply:
I was on the fence about this one. Two single fields that seem similar might not be an appropriate threshold to share. 🤔 

These types are exported, but we don't have an importable module... 🤷🏻

Reply:
I like it. Makes me think of other places like `spec.Metadata` and maybe even `PostgresClusterDataSource` (used in postgrescluster and pgbackrest)